tags existing deployment nodes as "found"

juanvallejo opened the pull request:
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1544183#c3
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1540560#c7
Followup to: https://github.com/openshift/origin/pull/18579

cc @soltysh

liggitt commented on pkg/oc/graph/kubegraph/nodes/nodes.go:
```diff
@@ -259,7 +259,7 @@ func EnsureDeploymentNode(g osgraph.MutableUniqueGraph, deployment *extensions.D
 		nodeName,
 		func(node osgraph.Node) graph.Node {
-			return &DeploymentNode{Node: node, Deployment: deployment}
+			return &DeploymentNode{Node: node, Deployment: deployment, IsFound: true}
```
why are we setting this to IsFound? I thought that only got set when we verified its existence in the API

juanvallejo replied:
We [set it to `false` in `FindOrCreateSynthetic...Node`](https://github.com/openshift/origin/blob/master/pkg/oc/graph/kubegraph/nodes/nodes.go#L103), when a synthetic node is created if a resource is not able to be retrieved / found. 

We appear to set IsFound to `true` here [in `Ensure...Node` funcs](https://github.com/openshift/origin/blob/master/pkg/oc/graph/kubegraph/nodes/nodes.go#L124)

liggitt replied:
not in &StatefulSetNode{node, statefulSet, false}...

juanvallejo replied:
> not in &StatefulSetNode{node, statefulSet, false}...

That appears to be a mistake I made in this PR:
https://github.com/openshift/origin/pull/18579/files#diff-5b718bf0b0ba776d85871d51d87dd74a

Ensure...Node functions are [called from projectstatus.go](https://github.com/openshift/origin/blob/master/pkg/oc/cli/describe/projectstatus.go#L1612) for [every node that is successfully loaded from the server](https://github.com/openshift/origin/blob/master/pkg/oc/cli/describe/projectstatus.go#L1600).

I have updated this PR to correct this.

Additionally, it appears that a `ScalingEdgeKind` [was not being added between hpa nodes and replicaSet nodes](https://github.com/openshift/origin/blob/master/pkg/oc/graph/kubegraph/edges.go#L259). I have updated this PR to address this as well.

juanvallejo commented:
/test extended_networking_minimal

juanvallejo commented:
/test cmd
/test unit

liggitt commented:
can you add a test that exercises this function that would have failed on the previous bugs?

juanvallejo commented:
@liggitt thanks, added a test.

Also, updated PR to address the last comment in https://bugzilla.redhat.com/show_bug.cgi?id=1540560#c7 which calls for displaying statefulsets and daemonsets as part of `oc status`

juanvallejo commented:
/retest

juanvallejo commented:
@liggitt friendly ping

liggitt commented:
/lgtm

juanvallejo commented:
/test extended_networking_minimal
